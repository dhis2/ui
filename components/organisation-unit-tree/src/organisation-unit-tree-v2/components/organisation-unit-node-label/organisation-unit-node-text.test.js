import { shallow } from 'enzyme'
import React from 'react'
import {
    getMatchingSegments,
    OrganisationUnitNodeText,
} from './organisation-unit-node-text.js'

describe('OrganisationUnitNodeText', () => {
    it('renders `displayName` when no `filteredString` is provided', () => {
        const wrapper = shallow(
            <OrganisationUnitNodeText
                isDisabled={false}
                toggleOpen={() => {}}
                displayName="the quick brown fox jumps over the lazy dog"
                filteredString=""
                isFilterMatch
            />
        )
        // 1 extra for styles
        expect(wrapper.children()).toHaveLength(2)
        expect(wrapper.find('span.highlight')).toHaveLength(0)
    })
    it('renders `displayName` when not `isFilterMatch`', () => {
        const wrapper = shallow(
            <OrganisationUnitNodeText
                isDisabled={false}
                toggleOpen={() => {}}
                displayName="the quick brown fox jumps over the lazy dog"
                filteredString="fox"
                isFilterMatch={false}
            />
        )
        // 1 extra for styles
        expect(wrapper.children()).toHaveLength(2)
        expect(wrapper.find('span.highlight')).toHaveLength(0)
    })
    it('renders `displayName` when `isFilterMatch`, `filteredString` not in `displayName`', () => {
        const wrapper = shallow(
            <OrganisationUnitNodeText
                isDisabled={false}
                toggleOpen={() => {}}
                displayName="the quick brown fox jumps over the lazy dog"
                isFilterMatch
                filteredString="aaaaaaaaaa"
            />
        )
        // 1 extra for styles
        expect(wrapper.children()).toHaveLength(2)
        expect(wrapper.find('span.highlight')).toHaveLength(0)
    })
    it('renders highlight spans when `isFilterMatch`, `filteredString` in `displayName`', () => {
        const wrapper = shallow(
            <OrganisationUnitNodeText
                isDisabled={false}
                toggleOpen={() => {}}
                displayName="the quick brown fox jumps over the lazy dog"
                isFilterMatch
                filteredString="fox"
            />
        )
        // 1 extra for styles
        expect(wrapper.children()).toHaveLength(4)
        expect(wrapper.find('span.highlight')).toHaveLength(1)
    })
    describe('getMatchingSegments', () => {
        it('handles a match at the middle of a string', () => {
            expect(
                getMatchingSegments(
                    'the quick brown fox jumps over the lazy dog',
                    'fox',
                    true
                )
            ).toMatchSnapshot()
        })
        it('handles a match at the start of a string', () => {
            expect(
                getMatchingSegments(
                    'the quick brown fox jumps over the lazy dog',
                    'the',
                    true
                )
            ).toMatchSnapshot()
        })
        it('handles a match at the end of a string', () => {
            expect(
                getMatchingSegments(
                    'the quick brown fox jumps over the lazy dog',
                    'dog',
                    true
                )
            ).toMatchSnapshot()
        })
        it('is case insensitive', () => {
            expect(
                getMatchingSegments(
                    'the quick brown FOX jumps over the lazy dog',
                    'fOx',
                    true
                )
            ).toMatchSnapshot()
        })
        it('treats spaces like regular characters', () => {
            expect(
                getMatchingSegments(
                    'the quick brown fox jumps over the lazy dog',
                    'quick brown fox',
                    true
                )
            ).toMatchSnapshot()
        })
        it('handles multiple matches', () => {
            expect(
                getMatchingSegments(
                    'the quick brown fox jumps over the lazy dog',
                    'the',
                    true
                )
            ).toMatchSnapshot()
        })
    })
})
