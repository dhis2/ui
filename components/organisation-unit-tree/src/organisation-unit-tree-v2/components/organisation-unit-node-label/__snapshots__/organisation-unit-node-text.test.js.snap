// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationUnitNodeText getMatchingSegments handles a match at the end of a string 1`] = `
Array [
  "the quick brown fox jumps over the lazy ",
  <span
    className="highlight"
  >
    dog
  </span>,
]
`;

exports[`OrganisationUnitNodeText getMatchingSegments handles a match at the middle of a string 1`] = `
Array [
  "the quick brown ",
  <span
    className="highlight"
  >
    fox
  </span>,
  " jumps over the lazy dog",
]
`;

exports[`OrganisationUnitNodeText getMatchingSegments handles a match at the start of a string 1`] = `
Array [
  <span
    className="highlight"
  >
    the
  </span>,
  " quick brown fox jumps over ",
  <span
    className="highlight"
  >
    the
  </span>,
  " lazy dog",
]
`;

exports[`OrganisationUnitNodeText getMatchingSegments handles multiple matches 1`] = `
Array [
  <span
    className="highlight"
  >
    the
  </span>,
  " quick brown fox jumps over ",
  <span
    className="highlight"
  >
    the
  </span>,
  " lazy dog",
]
`;

exports[`OrganisationUnitNodeText getMatchingSegments is case insensitive 1`] = `
Array [
  "the quick brown ",
  <span
    className="highlight"
  >
    FOX
  </span>,
  " jumps over the lazy dog",
]
`;

exports[`OrganisationUnitNodeText getMatchingSegments treats spaces like regular characters 1`] = `
Array [
  "the ",
  <span
    className="highlight"
  >
    quick brown fox
  </span>,
  " jumps over the lazy dog",
]
`;
